services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationData.kt

```diff
@@ -58,33 +69,22 @@ sealed interface ApplicationData {
 @JsonView(ChangeLogView::class)
 data class CableReportApplicationData(
     @JsonView(NotInChangeLogView::class) override val applicationType: ApplicationType,
-
-    // Common, required
-    override val name: String,
-    val customerWithContacts: CustomerWithContacts?,
-    override val areas: List<ApplicationArea>?,
-    val startTime: ZonedDateTime?,
-    val endTime: ZonedDateTime?,
     override val pendingOnClient: Boolean,
-
-    // CableReport specific, required
-    val workDescription: String,
-    val contractorWithContacts: CustomerWithContacts?, // työn suorittaja
-    val rockExcavation: Boolean?,
-
-    // Common, not required
+    override val name: String,
     val postalAddress: PostalAddress? = null,
-    val representativeWithContacts: CustomerWithContacts? = null, // Asianhoitaja
-    val invoicingCustomer: Customer? = null,
-    val customerReference: String? = null,
-    val area: Double? = null,
-
-    // CableReport specific, not required
-    val propertyDeveloperWithContacts: CustomerWithContacts? = null, // rakennuttaja
     val constructionWork: Boolean = false,
     val maintenanceWork: Boolean = false,
+    val propertyConnectivity: Boolean = false,
     val emergencyWork: Boolean = false,
-    val propertyConnectivity: Boolean = false, // tontti-/kiinteistöliitos
+    val rockExcavation: Boolean?,
+    val workDescription: String,
+    override val startTime: ZonedDateTime?,
```

Comment:
Why are the fields shuffled around? Is there a logic in the new order?

Reply:
The old way of having "common, required" etc did not make sense to me and was confusing and also not correct with excavation notification. The new order is based on the order in which the data is filled in Haitaton UI. At least for me it is much more clear now.

Reply:
In my opinion the order of the fields could follow the order of the UI forms. And it is better than using alphabetical ordering. This was not in my opinion an unnecessary change but made the code base better.